src/seedu/addressbook/logic/Logic.java

```diff
@@ -71,6 +71,7 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
     public CommandResult execute(String userCommandText) throws Exception {
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
+        saveAddressBook(command.isMutating());
```

Comment:
Since the behavior of this method (and the other execute method) has changed, the header comments should be updated accordingly.